README.md

```diff
@@ -147,6 +148,53 @@ You have these options to achieve that
    ViteExpress.listen(app, 3000, () => console.log("Server is listening..."));
     ```
 
+## 🤖 Transforming HTML
+
+You can specify transformer function that takes two arguments - HTML as a string and [`Request`][express-request] object - and apply any string transformation. Can be used to inject your custom metadata on the server-side. Needs to return transformed HTML in form of a string that will be served to the client.
+
+Imagine a situation in which your index.html file looks like this
+
+```html
+<html>
+   <meta>
+      <!-- placeholder -->
+   </meta>
+   <body>
+      <div id="root"></div>
+   </body>
+</html>
+```
+
+You can then use custom transformer function to replace the HTML comment with any string you like. It can be a custom meta tag. You can use request object to extract additional information about request such as requested page.
+
+```javascript
+import express from "express";
+import ViteExpress from "vite-express";
+
+function transformer(html: string, req: express.Request) {
+   return html.replace(
+      "<!-- placeholder -->", 
+      `<meta name="custom" content="${req.baseUrl}"/>`
+   )
+}
+
+ViteExpress.config({ transformer })
+ViteExpress.listen(express(), 3000);
```

Comment:
This `express()` looks weird. It's not clear to me if we can have middleware before transformer runs.

Reply:
Besides that, it looks good to me!

Reply:
Great point, it is a good idea to describe how this transformer function is working in relation to middlewares.

This express() thing is just a shorthand but I will remove it to avoid confusion. Thanks for your feedback @patryk-smc!